### ISSUE TITLE
[Custom Processors]  rust token_v2_processor missing most NFT burn events

### ISSUE THREAD

**Issue by flex-enjoy** [aptos-labs/aptos-indexer-processors#224]
## Description
If the NFT  mint & burn acttions are not too close in time, the burn event will be missing

## Repro
1. Run the token_v2_processor
2. Mint a NFT and wait for more than 1000 transaction versions
3. Burn the NFT
4. No burn event in `token_activities_v2` table for this NFT

## Analysis

The possible reason is the hashmap [token_v2_metadata_helper](https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/processors/token_v2_processor.rs#L639) is constructed by [write resource](https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/processors/token_v2_processor.rs#L530), but the burn event like [this](https://explorer.aptoslabs.com/txn/383064277/events?network=mainnet) has no such write resource, which breaks the process progress

**Comment by bowenyang007** [comment]
Good catch. Thanks for flagging! Looks like we need to reuse the same logic as token v2 ownership. https://github.com/aptos-labs/aptos-indexer-processors/blob/main/rust/processor/src/models/token_v2_models/v2_token_ownerships.rs#L222-L357. We'll need to fix this and backfill.
